lua/neogit/lib/git/cli.lua

```diff
@@ -709,6 +735,7 @@ local function new_builder(subcommand)
       env = state.env,
       pty = state.in_pty,
       verbose = verbose,
+      ignore_code = ignore_code,
```

Comment:
Oh, thats not a duplicate. I wound up implementing my own version of `on_error` because I didn't know there was an `on_error`. But `call_sync_ignoring_exit_code` and `call_ignoring_exit_code` (and the callers) need to be updated to use `external_errors` properly.

Reply:
Yes, not a duplicate per say, but a duplicate in functionality.

If you want to we can just leave it as is for now and I can merge those implementations together after merging everything, so that it is less work for you and you can keep focus on what's important

Reply:
Agreed, its due for a cleanup, and yea I'd like to just get this all merged and come back 😅 Even whats in PR's is already behind.. i made the revert popup two days ago, and added a force delete option for deleting unmerged branches last night...